### PR TITLE
Add shape inference for Upsample

### PR DESCRIPTION
**Pull request by philass**
Signed-off-by: Philip Lassen <plassen@groq.com>

### CODE REVIEW

File: src/Dialect/ONNX/ONNXOps.cpp [hunk 1]
```diff
@@ -5476,7 +5476,58 @@ LogicalResult ONNXUniqueOp::inferShapes(
 
 LogicalResult ONNXUpsampleOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
-  return emitError(NOT_IMPLEMENTED_MESSAGE);
+  // Sanity checks on input data argument
+  if (!X().getType().isa<RankedTensorType>()) {
```
Comment (chentong319) [inline src/Dialect/ONNX/ONNXOps.cpp:5480]: Could the sanity check be put into verifier?

File: src/Dialect/ONNX/ONNXOps.cpp [hunk 2]
```diff
@@ -5476,7 +5476,58 @@ LogicalResult ONNXUniqueOp::inferShapes(
...
+  if (!scalesConstOp) {
+    return success();
+  }
+  auto valueAttr = scalesConstOp.valueAttr().dyn_cast<DenseElementsAttr>();
+  if (!valueAttr) {
+    return emitError("Scales constant is not a DenseElementsAttr");
+  }
+  int scaleIdx = 0;
+  // Why are the scale values float's?
+  for (auto it = valueAttr.getValues<FloatAttr>().begin();
```
Comment (chentong319) [inline src/Dialect/ONNX/ONNXOps.cpp:5512]: It is possible that scale is the result of some ratio, which is a float.

**Comment by chentong319** [comment]
I saw upsample beams deprecated. Can we use onnx.resize to express onnx.upsample?

File: src/Dialect/ONNX/ONNXOps.cpp (hunk 1 above)
Comment (philass) [reply src/Dialect/ONNX/ONNXOps.cpp:5518]: Good point. I've added a verifier, and put the checks in there

File: src/Dialect/ONNX/ONNXOps.cpp (hunk 2 above)
Comment (philass) [reply src/Dialect/ONNX/ONNXOps.cpp:5548]: Ahh yes that makes sense. Thanks for the clarification

**Comment by philass** [comment]
@chentong319 thanks for the review,

I've updated the PR to address the comments. Please let me know if you would like any further changes

**Review by chentong319 — APPROVED** [review]
LGTM!

**Pull request merged**